examples/cosmos/ics20/ics20.qnt

```diff
@@ -480,5 +480,59 @@ module ics20Test {
       chainStates' = chainStates
   })
 
-  // TODO(thpani): add a test for packet timeout
+  // From
+  // https://github.com/cosmos/ibc-go/blob/a25f0d421c32b3a2b7e8168c9f030849797ff2e8/modules/apps/transfer/transfer_test.go#L44
+  run TransferTest = {
+    pure val amount = 9223372036854775808
+
+    // NOTE: fungible token is prefixed with the full trace in order to verify the packet commitment
+    pure val denomInA = ATOM
+    pure val denomSentFromAToB = denomInA.with("path", [{ port: "transfer", channel: "channelToA" }].concat(denomInA.path))
+    pure val denomSentFromBToC = denomSentFromAToB.with("path", [{ port: "transfer", channel: "channelToB" }].concat(denomSentFromAToB.path))
+    init.then(
+      // Make sure that Alice has enough tokens on chain A. I did not understand
+      // how this is done in the go test.
+      chainStates' = chainStates.setBy("A", state => state.with("bank", state.bank.setBy("alice", balances => balances.set(ATOM, amount))))
```

Comment:
`chainStates.setBy("A", state => state.with("bank", state.bank.setBy("alice", balances => balances.set(ATOM, amount))))`

I feel like this expression is asking for a helper definition that make this much shorter and understandable, similar to TLA+'s `EXCEPT` actually:

```
pure val setBalanceIn(chain, account, denom, amount) = {
  chainStates.setBy(chain, state =>
    state.with("bank", state.bank.setBy(account, balances => balances.set(denom, amount)))
  )
}
```

It's actually interesting to see how these calls compose. The functional style seems right to me.